cyrw/ioctl.go

```diff
@@ -81,9 +81,6 @@ func (d *Device) tx(packet []byte) (err error) {
 	const PADDING_SIZE = 2
 	totalLen := uint32(whd.SDPCM_HEADER_LEN + PADDING_SIZE + whd.BDC_HEADER_LEN + len(packet))
 
-	d.Lock()
```

Comment:
I'm assuming you intend to move lock up to Tx entry point (SendEth), such as:

```
func (d *Device) SendEth(pkt []byte) error {
        d.lock()
        defer d.unlock()
        return d.tx(pkt)                                                                                   
} 
```

Reply:
Yes-  so the rule for locking using this methodology as I understand it is:

- [x] is it exported or is it a async task that outlives it's exported caller
- [x] does it do I/O

If both boxes are ticked then lock it